CAMEL-13234: Remove dependencies weld1 and weld2

DenisIstomin opened the pull request:
Completely remove weld1 and weld2.

davsclaus commented on components/camel-cdi/pom.xml:
```diff
@@ -190,113 +190,11 @@
     <profiles>
         <profile>
-            <id>weld-1.0</id>
+            <id>weld-2.0</id>
```
Maybe call it weld3 since its weld 3 we use ;)

DenisIstomin replied:
Done

DenisIstomin commented:
1. Renamed profile weld-2.0 to weld-3.0 at components/camel-cdi/pom.xml
1. Renamed profile weld-2.0 to weld-3.0 at tests/camel-itest-cdi/pom.xml
There are no more occurrences of weld-2.0 profile name.
1. Removed 2 obsolete properties `cdi-api-1.0-version` and `cdi-api-1.1-version` at parent/pom.xml. They are no longer in use.
